Kernel/Devices/USB/USBTransfer.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <Kernel/Devices/USB/PacketTypes.h>
+#include <Kernel/Devices/USB/USBPipe.h>
+#include <Kernel/VM/PhysicalPage.h>
+#include <Kernel/VM/Region.h>
+
+// TODO: Callback stuff in this class please!
+namespace Kernel::USB {
+class USBTransfer : public RefCounted<USBTransfer> {
+public:
+    static NonnullRefPtr<USBTransfer> create(const USBPipe& pipe, u16 len);
```

Comment:
Since USB is hot plugable, it appears these could technically be called at any time during execution?
If that's true then we should be returning `RefPtr` and using `adopt_ref_if_nonnull`.

Reply:
This is a good general comment for the whole change I suppose.